chain/ethereum/src/block_ingestor.rs

```diff
@@ -1,9 +1,35 @@
+use std::collections::HashMap;
 use std::time::Duration;
 use std::time::Instant;
 
 use graph::prelude::*;
 use web3::types::*;
 
+pub struct BlockIngestorMetrics {
+    pub block_number: Box<GaugeVec>,
+}
+
+impl BlockIngestorMetrics {
+    pub fn new<M: MetricsRegistry>(registry: Arc<M>) -> Self {
```

Comment:
We've been able to get away with `Arc<dyn MetricsRegistry>` in a lot of places. This should work here as well; then we don't have to use generics.

Reply:
Why do we avoid using generics?

Reply:
I've been trying to avoid them where I could to simplify type/method signatures. Admittedly, I don't understand the drawbacks of either approach.

Reply:
The conclusion is that because this is function parameter the best syntax is `fn new(registry: Arc<impl MetricsRegistry>)`

Reply:
Hmmm.. `new(registry: Arc<impl MetricsRegistry>)` does not compile for `BlockIngestorMetrics` and `GraphQLServiceMetrics`. 

<img width="1125" alt="Screen Shot 2019-12-12 at 10 46 59" src="https://user-images.githubusercontent.com/1591297/70739947-d45d1e80-1ccc-11ea-82e6-bd6767829819.png">

Btw, the format I started with, using generics, matches all of the existing metrics struct implementations:
  - `SubgraphInstanceManagerMetrics`
  - `SubgraphInstanceMetrics`
  - `HostMetrics`
  - `ProviderEthRpcMetrics`
  - `SubgraphEthRpcMetrics`
  - `BlockStreamMetrics`

I've updated each of these  according to this conversation and they all successfully compile with `new(registry: Arc<impl MetricsRegistry>)`.

Reply:
![image](https://user-images.githubusercontent.com/19324/70750266-6b0df780-1d2e-11ea-9f77-6a2292ea3792.png)

Works fine for me?

Reply:
That compiles over here.

Reply:
But that's without the signature cleanup suggestions you made about the `BlockIngestor` struct right? ;)